crates/http-api-bindings/src/embedding/llama.rs

```diff
@@ -46,10 +46,45 @@ impl Embedding for LlamaCppEngine {
             request = request.bearer_auth(api_key);
         }
 
-        let response = request
-            .send()
-            .instrument(embedding_info_span!("llamacpp"))
-            .await?;
+        // Some initial requests to llama.cpp are experiencing issues
+        // would failed with `Connection reset by peer` or `Broken pipe`
+        //
+        // This serves as a temporary solution to attempt the request up to three times.
+        //
+        // Track issue: https://github.com/ggerganov/llama.cpp/issues/11411
+        let mut attempts = 0;
+        let max_attempts = 3;
+        let response = loop {
```

Comment:
use `ExponentialBackoff` from `tokio_retry`.

Reply:
If we're sure that this situation only happens in server start, consider create a timer to disable the retry behavior after server has been up for 2 hours.

Reply:
No, it occurs consistently after a period without any embedding requests, let me confirm this again later